mantidimaging/gui/windows/spectrum_viewer/test/model_test.py

```diff
@@ -193,6 +193,18 @@ def test_save_csv(self):
         self.assertIn("1.0,2.0,2.0", mock_stream.getvalue())
         self.assertTrue(mock_stream.is_closed)
 
+    def test_WHEN_save_csv_called_THEN_save_roi_coords_called_WITH_correct_args(self):
+        path = Path("test_file.csv")
+        with mock.patch('builtins.open', mock.mock_open()) as mock_open:
+            self.model.save_roi_coords(path)
+        mock_open.assert_called_once_with(path, encoding='utf-8', mode='w')
+
+    def test_WHEN_get_roi_coords_filename_called_THEN_correct_filename_returned(self):
+        mock_path = mock.create_autospec(Path)
+        mock_path.name = "test_file.csv"
+        expected_path = mock_path.with_stem(f"{mock_path.stem}_roi_coords")
+        self.assertEqual(expected_path, self.model.get_roi_coords_filename(mock_path))
```

Comment:
A real `Path` is safe to use in tests (you could use a PurePath to be sure).

I'd suggest something like
`self.assertEqual(Path("test_roi_coords.csv"), self.model.get_roi_coords_filename(Path("test.csv")))`